openapi-core/src/main/java/io/ballerina/openapi/core/GeneratorUtils.java

```diff
@@ -1037,4 +1044,51 @@ private static SemanticModel getSemanticModel(Path clientPath) throws ProjectExc
             throw new ProjectException(e.getMessage());
         }
     }
+
+    public static String getOpenAPIType(Schema<?> schema) {
+        if (schema.getTypes() != null && !schema.getTypes().isEmpty()) {
+            for (String type : schema.getTypes()) {
```

Comment:
Are we going to set the `nullable` details here, If there is a `null` in the type list?

Reply:
Yes. In schema generation, type `null` is considered at [https://github.com/ballerina-platform/openapi-tools/blob/8d7d82fca3c36c0a195297ae835b3db6b86668b8/openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java#L149](https://github.com/ballerina-platform/openapi-tools/blob/8d7d82fca3c36c0a195297ae835b3db6b86668b8/openapi-core/src/main/java/io/ballerina/openapi/core/generators/schema/TypeGeneratorUtils.java#L149). In parameter generation `null` does not hold any meaning. 

*   When the parameter is `required: true` the value cannot be `null`
*   When the parameter is optional but has a default value, in the case of user does not pass any value we send the default value
*   When parameter is optional and has no default value we generate a nillable parameter despite of the `null` configured.